### ISSUE TITLE
Gutter utility overrides margins in all directions

### ISSUE THREAD

**Issue by conormuirhead** [ThinkUniform/uniformcss#17]
**Describe the bug**
The `.gutter-x-*` and `.gutter-y-*` utility class names imply that they'll only affect either vertical or horizontal spacing between elements. Unfortunately, their output is set up to apply margin on all sides, setting a value of `0` on the sides it's not directly setting:

```
.gutter-y-1 > * + * {
  --gutter-top: 1;
  --gutter-bottom: 0;
  margin: calc(0.0625rem * var(--gutter-top)) 0 calc(0.0625rem * var(--gutter-bottom)) 0;
}
```

This means that any direct children of a `gutter` with additional margin utilities for the other sides will not take effect.

**To Reproduce**
1. Apply `.gutter-y-16` to a parent element with 2+ direct children
2. Apply `.mx-16` to one of the direct children
4. See that the child element's `.mx-16` margin is overridden by the overly broad `margin` declaration on `gutter`

**Expected behavior**
I would expect that `.gutter-y-*` would output only `margin-top` and `margin-bottom`. And likewise that `x` would only output `margin-left` and `margin-right`. Something more like this:

```
.gutter-y-1 > * + * {
  --gutter-top: 1;
  --gutter-bottom: 0;
  margin-top: calc(0.0625rem * var(--gutter-top)); 
  margin-bottom: calc(0.0625rem * var(--gutter-bottom));
}
```

**Comment by jinsupark** [comment]
Yeah, this one was specifically designed this way because of how the utility generator works. If we wanted to utilize the automatic sizing map range selection whilst still implementing the gutter-reverse functionality we may have to come up with a different solution to how this utility is generated!

I'll keep this in mind!

**Comment by jinsupark** [comment]
Hey @conormuirhead this issue is now addressed in **version 1.4.0** alongside other changes! 👍

**Comment by conormuirhead** [comment]
Awesome, thanks!
